pkg/apis/storage/validation/validation.go

```diff
@@ -163,6 +163,12 @@ func validateVolumeAttachmentSource(source *storage.VolumeAttachmentSource, fldP
 	if source.PersistentVolumeName == nil || len(*source.PersistentVolumeName) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath, ""))
 	}
+	if source.PersistentVolumeName != nil {
+		for _, msg := range apivalidation.ValidatePersistentVolumeName(*source.PersistentVolumeName, false) {
```

Comment:
same comment about only adding this validation in volumeAttachmentStrategy#Validate, and only when the api request is not coming from the v1beta1 API

Reply:
I think this was fixed by commit below.